File: woodstorage/src/main/java/com/jordylangen/woodstorage/view/TagFilterContract.java
package com.jordylangen.woodstorage.view;

import java.util.List;

import rx.Observable;

public interface TagFilterContract extends Contract {

    interface View extends Contract.View {

        void add(SelectableTag selectableTag);

        void addAll(List<SelectableTag> selectedTags);
    }

    interface Presenter extends Contract.Presenter<View> {

        void tagSelectedChanged(SelectableTag selectableTag, boolean isChecked);

        Observable<List<SelectableTag>> observeSelectedTags();
    }
}

File: woodstorage/src/main/java/com/jordylangen/woodstorage/view/TagFilterContract.kt
package com.jordylangen.woodstorage.view

import io.reactivex.Observable

interface TagFilterContract : Contract {

    interface View : Contract.View {

        fun add(selectableTag: SelectableTag)

        fun set(selectedTags: List<SelectableTag>)
    }

    interface Presenter : Contract.Presenter<View> {

        fun tagSelectedChanged(selectableTag: SelectableTag, isChecked: Boolean)

        fun observeSelectedTags(): Observable<List<SelectableTag>>
    }
}
